ValetTouchIDTest/ValetTouchIDTestViewController.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  ValetTouchIDTestViewController.swift
+//  Valet
+//
+//  Created by Eric Muller on 4/20/16.
+//  Copyright © 2016 Square, Inc. All rights reserved.
+//
+
+import UIKit
+
+
+final class ValetTouchIDTestViewController : UIViewController
+{
+    // MARK: Properties
+
+    @IBOutlet var textView : UITextView?
+    let secureEnclaveValet : VALSecureEnclaveValet = VALSecureEnclaveValet.init(identifier: "UserPresence", accessControl: VALAccessControl.UserPresence) as VALSecureEnclaveValet!
+    let username : String = "CustomerPresentProof"
+
+
+    // MARK: Outlets
+
+    @IBAction func setOrUpdateItem(sender: UIResponder)
+    {
+        let stringToSet = "I am here! " + NSUUID.init().UUIDString
+        let setOrUpdatedItem = secureEnclaveValet.setString(stringToSet, forKey: username)
+        updateTextView([#function, (setOrUpdatedItem ? "Success" : "Failure")])
+    }
+
+    @IBAction func getItem(sender: UIResponder)
+    {
+        var userCancelled: ObjCBool = false
```

Comment:
Good question! We need to communicate to the caller the difference between a user cancelling Touch ID and the item not being in the keychain. Unfortunately, due to bugs in Apple's code, `containsObjectForKey:` isn't guaranteed to give us the right information (see #67 for a detailed discussion); the only way to confirm whether a secure enclave object exists in the keychain is to attempt to fetch it with Touch ID.

In ObjC land, an inout bool was the best approach. In a Swift world, I think we'll want to turn the result into an enum with associated values. @EricMuller22 and I are going to discuss this on Monday at 3:30pm. LMK if you'd like to be part of that meeting :)